src/main/java/bio/terra/cli/command/resource/addref/GcsObject.java

```diff
@@ -41,10 +52,29 @@ protected void execute() {
     // build the resource object to add
     CreateResourceParams.Builder createResourceParams =
         referencedResourceCreationOptions.populateMetadataFields();
+
+    // parsing the path as bucket name and object name
+    if (gcsPath != null) {
+      if (bucketName != null || objectName != null) {
+        throw new UserActionableException("Specify only one path to add reference.");
+      } else {
+        Pattern r = Pattern.compile("(?:^gs://)([^/]*)/(.*)");
+        Matcher m = r.matcher(gcsPath);
+        if (m.find()) {
```

Comment:
what happens if this find returns false?  i don't see an exception?

you should probably have a test for an invalid --gcs-path format.